tutorial/tests/test_object_oriented_programming.py

```diff
@@ -2,6 +2,219 @@
 
 import pytest
 
+#
+# Example 1: Person
+#
+
+
+def reference_oop_person(first_name: str, last_name: str):
+    class Person:
+        """A class representing a person with first name and last name"""
+
+        def __init__(self, first_name: str, last_name: str):
+            self.first_name = first_name
+            self.last_name = last_name
+
+    return Person(first_name, last_name)
+
+
+def validate_oop_person(solution_result):
+    assert not isinstance(
+        solution_result, (str, int, float, bool, list, dict, tuple)
+    ), "Solution must return a class instance, not a primitive type."
+    assert (
+        type(solution_result).__module__ != "builtins"
+    ), "Solution must return an instance of a custom class, not a built-in type."
+    assert (
+        type(solution_result).__name__ == "Person"
+    ), "The class should be named 'Person'."
+    # Check the class attributes
+    try:
+        attrs = list(vars(solution_result))
+    except TypeError as e:
+        raise AssertionError from e

```

Comment:
If we `raise from`, pytest will catch the `TypeError` first, which will give the output "Your code could not run because of an error". If we **don't want** this, simply raise an `AssertionError` with a message.

Reply:
This will happen in the simplest case: if the functions returns `None`. So we should add an assertion to exclude a `None` return value. I know it's dumb, but we want to avoid that exception.

Reply:
I was struggling with the pre-commit checks because they did not like custom messages when raising an `AssertionError` and this was the easiest workaround. I will try to figure it out.